hibernate-core/src/main/java/org/hibernate/query/sqm/tree/select/SqmSelectStatement.java

```diff
@@ -113,6 +118,27 @@ private SqmSelectStatement(
 		this.parameters = parameters;
 	}
 
+	/**
+	 * A query that returns the number of results of this query.
+	 *
+	 * @since 6.5
+	 */
+	@Incubating
+	public SqmSelectStatement<Long> countQuery() {
+		final SqmSelectStatement<?> copy = copy( noParamCopyContext() );
+		final SqmQuerySpec<?> querySpec = copy.getQuerySpec();
+		for ( SqmRoot<?> root : querySpec.getRootList() ) {
+			final List<SqmJoin<?, ?>> joins = root.getOrderedJoins();
+			if ( joins != null && !joins.isEmpty() ) {
+				joins.removeIf(join -> join.getSqmJoinType() == SqmJoinType.LEFT);
```

Comment:
Left joins that join against a non-unique target column i.e. *-to-many joins affect the result count, so you can't just remove these joins.